### PR TITLE
oribital velocity

### PR DESCRIPTION
**Pull request by mikecoker**
random seed override

Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/particle/animator/velocity-overtime.ts [hunk 1]
```diff
@@ -54,7 +56,7 @@ export default class VelocityOvertimeModule extends ParticleModuleBase {
      * @zh 是否启用。
      */
     @displayOrder(0)
-    public get enable (): boolean {
+    public get enable () {
```
Comment (hulcypdiscord) [inline cocos/particle/animator/velocity-overtime.ts]: why did you remove the return type?

File: cocos/particle/animator/velocity-overtime.ts [hunk 2]
```diff
@@ -159,9 +231,39 @@ export default class VelocityOvertimeModule extends ParticleModuleBase {
...
             Vec3.transformQuat(vel, vel, this.rotation);
         }
-        Vec3.add(p.animatedVelocity, p.animatedVelocity, vel);
+        // add linear velocity
+        const speed = this.speedModifier.evaluate(normalizedTime, pseudoRandom(p.randomSeed + VELOCITY_X_OVERTIME_RAND_OFFSET))!;
+        Vec3.add(p.animatedVelocity, p.animatedVelocity, vel.multiplyScalar(speed));
+
+        // calculate orbital velocity
+        const offX = this.offsetX.evaluate(normalizedTime, pseudoRandom(p.randomSeed ^ VELOCITY_X_OVERTIME_RAND_OFFSET))!;
+        const offY = this.offsetY.evaluate(normalizedTime, pseudoRandom(p.randomSeed ^ VELOCITY_X_OVERTIME_RAND_OFFSET))!;
```
Comment (hulcypdiscord) [inline cocos/particle/animator/velocity-overtime.ts]: Should these use the correct axis variants of these?

File: cocos/particle/animator/velocity-overtime.ts [hunk 3]
```diff
@@ -159,9 +231,39 @@ export default class VelocityOvertimeModule extends ParticleModuleBase {
         if (this.needTransform) {
             Vec3.transformQuat(vel, vel, this.rotation);
         }
-        Vec3.add(p.animatedVelocity, p.animatedVelocity, vel);
+        // add linear velocity
+        const speed = this.speedModifier.evaluate(normalizedTime, pseudoRandom(p.randomSeed + VELOCITY_X_OVERTIME_RAND_OFFSET))!;
```
Comment (hulcypdiscord) [inline cocos/particle/animator/velocity-overtime.ts]: It looks like previously the pseudo rand value was only used if `isCurveTwoValues` was true, via using `randSpeed`

**Review by hulcypdiscord — COMMENTED** [review]
Let's hold off on this until post demo please...

File: cocos/particle/animator/velocity-overtime.ts (hunk 1 above)
Comment (mikecoker) [reply cocos/particle/animator/velocity-overtime.ts]: actually I didn't Not sure what happened here. prettier?

Comment (mikecoker) [reply cocos/particle/animator/velocity-overtime.ts]: or maybe it was not in the older engine version.

File: cocos/particle/animator/velocity-overtime.ts (hunk 3 above)
Comment (mikecoker) [reply cocos/particle/animator/velocity-overtime.ts]: The randSpeed stuff was not actually used atm. All of this came from the 3.7.3 branch I did a really long time ago so I'm kinda fuzzy on the details. It does appear to work though.
